packages/node-core/src/indexer/poi/poi.service.ts

```diff
@@ -159,10 +160,67 @@ export class PoiService implements OnApplicationShutdown {
         },
       },
     });
+
+    await batchDeletePoi(this.poiRepo.model, transaction, targetBlockHeight);
     const lastSyncedPoiHeight = await this.storeCache.metadata.find('latestSyncedPoiHeight');
+
     if (lastSyncedPoiHeight !== undefined && lastSyncedPoiHeight > targetBlockHeight) {
-      this.storeCache.metadata.set('latestSyncedPoiHeight', targetBlockHeight);
+      const genesisPoi = await this.poiRepo.model.findOne({
+        order: [['id', 'ASC']],
+        transaction: transaction,
+      });
+      // This indicates reindex height is less than genesis poi height
+      // And genesis poi has been remove from `batchDeletePoi`
+      if (!genesisPoi) {
+        this.storeCache.metadata.bulkRemove(['latestSyncedPoiHeight']);
+      } else {
+        this.storeCache.metadata.set('latestSyncedPoiHeight', targetBlockHeight);
+      }
     }
     this.storeCache.metadata.bulkRemove(['lastCreatedPoiHeight']);
   }
 }
+
+// REMOVE 10,000 record per batch
+async function batchDeletePoi(
+  model: PoiRepo,
+  transaction: Transaction,
+  targetBlockHeight: number,
+  batchSize = 10
+): Promise<void> {
+  // const offset = 0;
+  let completed = false;
+  // eslint-disable-next-line no-constant-condition
+  while (!completed) {
```

Comment:
Why not use `sqlIterator` there? It seems expensive to load before removing.

This maybe useful. Static destroy returns the number of items destroyed https://sequelize.org/api/v6/class/src/model.js~model#static-method-destroy

I see the same is happening with other data.

Reply:
sqlIterator apply sql to whole table, it is useful for sql like `create index xxx` for all records.  

In batchDeletePoi, we only need to remove records with certain block ranges. Compare to sqlIteraor, even we add `where`  conditions to excute the block range in `sql`, it will still iterate the complete table. I think this is the better approach.